docs/src/configuration/routes/_index.md

```diff
@@ -1,348 +1,337 @@
-    type: upstream
-    upstream: "app:http"
-"https://blog.example.com/":
-    type: upstream
-    upstream: "blog:http"
-```
+It also works on development environments.
+If you have a `feature` branch, it's `{default}` domain looks something like:
+`feature-def123-vmwklxcpbi6zq.us.platform.sh` (depending on the project's region).
+So requests to `blog.feature-def123-vmwklxcpbi6zq.us.platform.sh` and `us.feature-def123-vmwklxcpbi6zq.eu.platform.sh`
+are both routed to the same endpoint.
 
-In this case, there are two application containers `app` and `blog`.
-In a production environment, they would be accessible at `www.example.com` and `blog.example.com`, respectively.
-On a development branch named `sprint`, however, they would be accessible at URLs something like:
+Let's Encrypt wildcard certificates aren't supported (they would need DNS validation).
+So if you want to use a wildcard route and protect it with HTTPS,
+you need to provide a [custom TLS certificate](../../domains/steps/tls.md).
 
-```bash
-https://www.example.com.sprint-7onpvba-tvh56f275i3um.eu-2.platformsh.site/
-https://blog.example.com.sprint-7onpvba-tvh56f275i3um.eu-2.platformsh.site/
-```
+{{< note >}}
 
-If your project involves only a single apex domain with one app or multiple apps under subdomains,
-it's generally best to use the `{default}` placeholder.
-If you are running [multiple applications](/configuration/app/multi-app.md) on different apex domains,
-then you will need to use a static domain for all but one of them.
+In projects created before November 2017, the `.` in subdomains was replaced with a triple hyphen (`---`).
+It was switched to preserve `.` to simplify SSL handling and improve support for longer domains.
+If your project was created before November 2017, it still uses `---` to the left of the environment name.
+If you wish to switch to dotted-domains, please file a support ticket and we can do that for you.
+Doing so may change the domain name that your production domain name should CNAME to.
 
-Note that when there are two routes sharing the same HTTP scheme, domain, and path,
-where the first route is using the `{default}` placeholder and the other is using the `{all}` placeholder,
-the route using `{default}` takes precedence.
+{{< /note >}}
 
 ## Route identifiers
 
-All routes defined for an environment are available to the application in its `PLATFORM_ROUTES` environment variable,
-which contains a base64-encoded JSON object.
-This object can be parsed by the language of your choice to give your application access to the generated routes.
+When routes are generated, all placeholders (`{default}` and `{all}`) are replaced with appropriate domain names
+and any additional routes (such as redirecting HTTP to HTTPS) are created.
+This means the final generated routes differ by environment and so shouldn't be hard coded in your app.
+These routes are available in the `PLATFORM_ROUTES` environment variable as a base64-encoded JSON object.
```

Comment:
I was trying to refer to "final generated routes" specifically. Since they're different from the routes that are defined in the `routes.yaml` file and I thought that might be confusing.

Reply:
I see, I think the antecedent isn't very clear but the overall context is probably understood in both cases.